lib/config/config-file.js

```diff
@@ -360,8 +360,9 @@ function resolve(filePath) {
             var scopedPackageShortcutRegex = /^(@[^\/]+)(?:\/(?:eslint-config)?)?$/;
             if (scopedPackageShortcutRegex.test(filePath)) {
                 filePath = filePath.replace(scopedPackageShortcutRegex, "$1/eslint-config");
-            } else if (filePath.split("/")[1].indexOf("eslint-config-") !== 0) {
-                // for scoped packages, insert the eslint-config after the first /
+            } else if (!/^eslint-config(-|$)/.test(filePath.split("/")[1])) {
```

Comment:
By default, you'd use `@something` or `@something/eslint-config` to point to that package's "main" file.